src/main/java/org/hawkular/client/android/backend/BackendClient.java

```diff
@@ -0,0 +1,58 @@
+package org.hawkular.client.android.backend;
+
+import android.app.Activity;
+import android.support.annotation.NonNull;
+
+import org.hawkular.client.android.backend.model.Tenant;
+import org.jboss.aerogear.android.pipe.LoaderPipe;
+import org.jboss.aerogear.android.pipe.PipeManager;
+import org.jboss.aerogear.android.pipe.rest.RestfulPipeConfiguration;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public final class BackendClient
+{
+	public static final class Pipes
+	{
+		private Pipes() {
+		}
+
+		public static final String TENANTS = "tenants";
+	}
+
+	public static final class PipePaths
+	{
+		private PipePaths() {
+		}
+
+		public static final String TENANTS = "hawkular-metrics/tenants";
+	}
+
+	private String serverUrl;
+
+	public void setServerUrl(@NonNull String serverUrl) {
```

Comment:
As far as I see it user will be able to change server URL while running the application.

My initial idea was to make the `BackendClient` as a singleton so it can have state across activities. Otherwise it should be recreated by every activity and, as a result, all pipes will be recreated.